yktrace/src/tir.rs

```diff
@@ -119,13 +123,50 @@ impl TirTrace {
             // When adding statements to the trace, we clone them (rather than referencing the
             // statements in the SIR) so that we have the freedom to mutate them later.
             let user_bb_idx_usize = usize::try_from(loc.bb_idx).unwrap();
-            ops.extend(
-                body.blocks[user_bb_idx_usize]
-                    .stmts
-                    .iter()
-                    .cloned()
-                    .map(TirOp::Statement)
-            );
+            // When converting the SIR trace into a TIR trace we turn it into SSA form by alpha
+            // renaming all `Local`s and storing a mapping from old to new name in `rename_map`.
+            // This is a bit tricky when it comes to inlined function calls, which assume arguments
+            // start at `Local(1)`, which may have already been used. We thus need to store
+            // multiple levels in `rename_map` and have an additional variable `cur_call_args`,
+            // which provides information about which `Local`s we need to remove again from
+            // `rename_map` once we leave the inlined function call again (since we won't get
+            // `StorageDead` statements for arguments).
+            for stmt in body.blocks[user_bb_idx_usize].stmts.iter() {
+                let op = match stmt {
+                    Statement::StorageLive(local) => {
+                        let newlocal = Local(var_count);
+                        if let Some(v) = rename_map.get_mut(local) {
```

Comment:
It's a minor matter but this might be shorter as:

```rust
match rename_map.get_mut(local) {
    Some(v) => v.push(newlocal),
    None => rename_map.insert(local.clone(), vec![newlocal])
}
```

Reply:
Fixed in a62443d